src/DynamoCore/Extensions/IExtension.cs

```diff
@@ -50,5 +52,20 @@ public interface IExtension : IDisposable
         /// Action to be invoked when shutdown has begun.
         /// </summary>
         void Shutdown();
+
+        /// <summary>
+        /// Scan the PackagesDirectory for packages and attempt to load all of them.
+        /// </summary>
+        /// <param name="preferences">IPreferences instance, which contains 
+        /// a list of packages used by the Package Manager to determine, 
+        /// which packages are marked for deletion.</param>
+        /// <param name="pathManager">IPathManager instance 
+        /// with set of directories paths needed for Package Manager</param>
+        void Load(IPreferences preferences, IPathManager pathManager);
+
+        /// <summary>
+        /// Event which is fired, when it needs to load an assembly with nodes
+        /// </summary>
+        event Action<Assembly> RequestLoadNodeLibrary;
```

Comment:
Thanks @pboyer. Hi @marimano, I have created [a new internal task for you](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7847) to work on, please have a look at it and schedule it accordingly among your work items, thanks! :)